zebra-network/src/peer_set/candidate_set.rs

```diff
@@ -379,33 +379,32 @@ fn validate_addrs(
 /// This will consider all addresses as invalid if trying to offset their
 /// `last_seen` times to be before the limit causes an underflow.
 fn limit_last_seen_times(addrs: &mut Vec<MetaAddr>, last_seen_limit: DateTime32) {
-    let (oldest_reported_seen_timestamp, newest_reported_seen_timestamp) =
-        addrs
-            .iter()
-            .fold((u32::MAX, u32::MIN), |(oldest, newest), addr| {
-                let last_seen = addr
-                    .untrusted_last_seen()
-                    .expect("unexpected missing last seen")
-                    .timestamp();
-                (oldest.min(last_seen), newest.max(last_seen))
-            });
+    let last_seen_times = addrs.iter().map(|meta_addr| {
+        meta_addr
+            .untrusted_last_seen()
+            .expect("unexpected missing last seen: should be provided by deserialization")
+    });
+    let oldest_seen = last_seen_times.clone().min().unwrap_or(DateTime32::MIN);
+    let newest_seen = last_seen_times.max().unwrap_or(DateTime32::MAX);
 
     // If any time is in the future, adjust all times, to compensate for clock skew on honest peers
-    if newest_reported_seen_timestamp > last_seen_limit.timestamp() {
-        let offset = newest_reported_seen_timestamp - last_seen_limit.timestamp();
+    if newest_seen > last_seen_limit {
+        let offset = newest_seen
+            .checked_duration_since(last_seen_limit)
+            .expect("unexpected underflow: just checked newest_seen is greater");
```

Comment:
Would it make sense to make this:

```suggestion
    if let Some(offset) = newest_seen.checked_duration_since(last_seen_limit) {
```

but then I think the case for `newest_seen == last_seen_limit` might also end up having an offset of zero applied. Not sure if the conciseness is worth the extra performance penalty in some cases :thinking:

Reply:
I think it's fine either way.

This kind of code isn't important enough to benchmark. So I'm more concerned about clear code than performance.

Also, in this specific case, Rust optimises out the redundant expect, so there is no performance penalty at all:
```sh
$ strings ~/bin/zebrad-release | rg "just checked newest_seen is greater"

```